integration/storagetest/maptests.go

```diff
@@ -193,3 +196,64 @@ func (*mapTests) TestTileRoundTrip(ctx context.Context, t *testing.T, ms storage
 		t.Fatalf("ReadWriteTransaction(): %v", err)
 	}
 }
+
+func (*mapTests) TestGetSetRoundTrip(ctx context.Context, t *testing.T, ms storage.MapStorage, as storage.AdminStorage) {
+	tree := mustCreateTree(ctx, t, as, storageto.MapTree)
+	mustSignAndStoreMapRoot(ctx, t, ms, tree, &types.MapRootV1{Revision: uint64(0)})
+
+	num := int64(10)
+	leafRevs := make(map[int64]*trillian.MapLeaf)
+
+	// First create several revisions of a given map leaf and store them:
+	for rev := int64(1); rev <= num; rev++ {
+		index := sha256.Sum256([]byte("Key"))
+		value := []byte{byte(rev)}
+		leafHash := sha256.Sum256(value)
+		leaf := &trillian.MapLeaf{
+			Index:     index[:],
+			LeafValue: value,
+			LeafHash:  leafHash[:],
+			ExtraData: []byte{byte(rev)},
+		}
+		leafRevs[rev] = leaf
+
+		if err := ms.ReadWriteTransaction(ctx, tree, func(ctx context.Context, tx storage.MapTreeTX) error {
+			if err := tx.Set(ctx, index[:], leaf); err != nil {
+				t.Errorf("%d: Set() = %v, want no error", rev, err)
+				return err
+			}
+			return nil
+		}); err != nil {
+			t.Errorf("%d: ReadWriteTransaction() = %v, want no error", rev, err)
+			continue
+		}
+		mustSignAndStoreMapRoot(ctx, t, ms, tree, &types.MapRootV1{
+			Revision:       uint64(rev),
+			TimestampNanos: uint64(time.Now().UnixNano()),
+		})
+	}
+
+	// Now we try to read the different revisions out in a random order:
+	for rev, want := range leafRevs {
+		t.Logf("reading @ revision %d", rev)
+		tx, err := ms.SnapshotForTree(ctx, tree)
+		if err != nil {
+			t.Fatalf("SnapshotForTree(): %v", err)
+		}
+		leaves, err := tx.Get(ctx, rev, [][]byte{want.Index})
+		if err != nil {
+			t.Fatalf("Get(): %v", err)
+		}
+		if got, want := len(leaves), 1; got != want {
+			t.Errorf("Get(rev: %d): %d leaves, want %d", rev, got, want)
+			continue
+		}
+		if got := leaves[0]; !cmp.Equal(got, want, protocmp.Transform()) {
+			t.Errorf("Get(rev: %d) = diff:\n%v", rev, cmp.Diff(got, want, protocmp.Transform()))
+			continue
+		}
+		if err := tx.Commit(ctx); err != nil {
```

Comment:
The commit here was surprising in a read-only test, but worth testing it doesn't blow  up I guess.

Reply:
Hmm, good point. At the very least it should be outside the loop